### DISCUSSION TITLE
For one page

### DISCUSSION THREAD

**Discussion by konradbjk** [vercel/next.js#45253] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.6.0: Sun Nov  6 23:31:16 PST 2022; root:xnu-8020.240.14~1/
Binaries:
      Node: 16.18.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: N/
Relevant packages:
      next: 12.3.4
      eslint-config-next: 12.3.4
      react: 18.2.0
      react-dom: 18.2.0

### Which area(s) of Next.js are affected? (leave empty if unsure)

Static HTML Export (next export)

### Link to the code that reproduces this issue

---

### To Reproduce

Starting a project with `yarn dev` or exporting it and putting as a static page

### Describe the Bug

Even after I add the below meta line to my head tag for the `_app` or individual pages, there is some weird empty space on the side. Interestingly, when I check in the dev tools, this space seems not to be a part of the HTML element.
<img width="957" alt="Screenshot 2023-01-25 at 09 55 38" src="https://user-images.githubusercontent.com/31480935/214521523-785f9738-4e7f-4c82-a77b-ba185b2216ae.png">
<img width="928" alt="Screenshot 2023-01-25 at 09 56 00" src="https://user-images.githubusercontent.com/31480935/214521537-94f1103a-7d16-46cc-9d63-2befd7a5c0fc.png">

```html
<meta name="viewport" content="width=device-width, initial-scale=1" />
```

Initially, the page is zoomed in so that this space is not visible, but one can scroll through it or zoom out. I tried adding the below overscroll-behavior to global CSS, but it did not help; still, scrolling to the side and zooming out are possible.
```css
// global.css
@tailwind base;
@tailwind components;
@tailwind utilities;

html,
body {
  padding: 0;
  margin: 0;
  overscroll-behavior: none;
  background-color: #000000;
}
```

I am not able to replicate the same issue on any other pages that I have created. 

### Expected Behavior

The page is built properly

### Which browser are you using? (if relevant)

Firefox

### How are you deploying your application? (if relevant)

Cloudflare Pages, yarn dev

**Comment by icyJoseph** [comment]
## Edit - Solution

Your `testimonials` and `inspiration` sections have padding left and right, but also have a child that applies width `100vw`.

![Screenshot 2023-01-25 at 15 36 17](https://user-images.githubusercontent.com/21013447/214591493-3b8e6114-352c-4ced-a23b-678577cfe74b.png)


Notice how the palevioletred of that section bleeds out, compared to the section above. The issue is the usage of `w-screen` on a child of those sections. 

I think the easiest is to not apply padding to `testimonials` and `inspiration` when the screen is so narrow.

**Reply by icyJoseph** [reply]
@konradbjk Could you confirm that this works?

**Reply by konradbjk** [reply]
nope, it did not fix the problem. The padding is crucial for the carousel buttons. I changed the width to w-full

what fixes it, when I set the overflow-hidden in _app but I kind of not understand what cases this overflow if it is neither in body nor even HTML tag,

**Reply by icyJoseph** [reply]
You are a bit confused. The element that has width 100vw will push it's parent to contain it. Because the parent also applies padding then the entire thing overflows the screen width. It's not Next.js, nor tailwind, it's a CSS issue.

**Reply by icyJoseph** [reply]
Maybe you misunderstood where to change the padding here's a video. I removed the testimonials to reduce the problem and show the exact fix:

- padding-inline overrides -left and -right, but you just need to remove them for this screen widths
- first there's overflow-x
- apply the padding, and the overflow-x is gone
- The navigation arrows stay where they are supposed to be

https://user-images.githubusercontent.com/21013447/214668318-523872e0-589d-4c67-904a-4fa0ab2cbcc7.mov

And both fixed:


https://user-images.githubusercontent.com/21013447/214669363-ed5d6899-0253-48c8-8f5b-b757cc5e471f.mov

Most importantly, the arrows are not moved at all.

**Reply by konradbjk** [reply]
What I meant that the w-screen did nothing in this scenario. It seems that it is not forcing the screen size, I tried to force a screensize on another element without luck.

Also, once I add this `padding-inline`, the overflow does not disappear for me on Firefox. If I understand correctly, padding-inline is supposed to then make the carousel buttons go to left-0 and right-0?

**Reply by icyJoseph** [reply]
No, the padding-inline was just to remove the `padding-left` and `padding-right` in one line, but you gotta just do that yourself. 

I am afraid we are not communicating clearly here, but it seems like you've fixed it now?

**Reply by konradbjk** [reply]
@icyJoseph I did fix it by removing overflow on x-axis. I did 1-to-1 of what you said, but this did not fix the overflow for me. 
So padding-inline works like px-0 in tailwind?

Class | Properties
-- | --
px-0 | padding-left: 0px; padding-right: 0px;

I got lost in that CSS with those buttons. I am now keeping the basics from the tutorial. If I try to manipulate the padding, the buttons or text goes crazy on mobile, creating more overflow or super narrow.

I tried removing px in the top-level div, but no luck. I have tried some value variations, but it got only worse on mobile. I tried with padding for text, with smaller/bigger - no luck. Below you can find the CSS and jsx code. I migrate to tailwind layers slowly. What would you recommend here? 

```css
.embla__button {
  outline: 0;
  cursor: pointer;
  background-color: transparent;
  touch-action: manipulation;
  position: absolute;
  z-index: 1;
  top: 50%;
  transform: translateY(-50%);
  border: 0;
  width: 30px;
  height: 30px;
  justify-content: center;
  align-items: center;
  padding: 0;
}

.embla__button__svg {
  width: 100%;
  height: 100%;
}

@layer components {
  .embla__button--prev {
    @apply left-3;
  }

  .embla__button--next {
    @apply right-3;
  }
}
```

```js
<div id={name} className="py-10 px-2 text-center sm:px-6 lg:px-10">
      <MainHeading text="To możecie zrobić wspólnie" />
      <div
        className="relative mt-8 w-screen overflow-hidden px-2 sm:w-full sm:max-w-4xl sm:px-10"
        ref={viewportRef}>
        <div className="flex">
          {sliderElements.map((element) => (
            <div
              className="mx-2 flex w-full flex-none cursor-pointer flex-wrap items-center justify-center lg:flex-nowrap"
              key={element.id}>
              <div className="h-200 overflow-hidden">
                <img
                  src={element.icon}
                  height={180}
                  width={180}
                  className="object-contain"
                  alt="cover image"
                  placeholder="blur"
                />
              </div>
              {/* content */}
              <div className="flex max-w-md flex-col lg:max-w-prose">
                <p className="text-l px-6 text-justify text-gray-200 sm:pl-2 sm:pr-10">
                  {element.description}
                </p>
              </div>
            </div>
          ))}
        </div>
        <NextButton onClick={scrollNext} />
        <PrevButton onClick={scrollPrev} />
      </div>
    </div>
 ```

**Comment by guitheengineer** [comment, marked as answer]
This is probably some content overflow on the X axis. If you add `overflow-x: hidden;` to your body selector it will solve it, but also ideally you want to find what element is causing this and solve it there (edit: like icyJoseph did above).

**Reply by konradbjk** [reply]
Yes, to the point, but I have no idea how to find it. As you can see in the initial screenshots even the root element (HTML) is smaller. I imagine that the size of the page might be an issue too. @guilhermefront Do you have some tip on how to find the cause?
